src/vm/peimagelayout.cpp

```diff
@@ -259,20 +284,20 @@ void PEImageLayout::ApplyBaseRelocations()
 #ifndef CROSSGEN_COMPILE
     if (dwOldProtection != 0)
     {
-        BOOL bExecRegion = (dwOldProtection & (PAGE_EXECUTE | PAGE_EXECUTE_READ |
-            PAGE_EXECUTE_READWRITE | PAGE_EXECUTE_WRITECOPY)) != 0;
-
         // Restore the protection
         if (!ClrVirtualProtect(pWriteableRegion, cbWriteableRegion,
                                dwOldProtection, &dwOldProtection))
             ThrowLastError();
-
-        if (bRelocDone && bExecRegion)
-        {
-            ClrFlushInstructionCache(pWriteableRegion, cbWriteableRegion);
-        }
     }
 #endif // CROSSGEN_COMPILE
+
+    BOOL bExecRegion = (dwOldProtection & (PAGE_EXECUTE | PAGE_EXECUTE_READ |
+            PAGE_EXECUTE_READWRITE | PAGE_EXECUTE_WRITECOPY)) != 0;
+
+    if (bRelocDone && bExecRegion)
+    {
+        ClrFlushInstructionCache(pFlushRegion, cbFlushRegion);
```

Comment:
I think this may miss flushing the very last page that we have written the relocs into. Nothing seems to check `bCrossPageBoundary` if it gets set on the very last page.

Reply:
Done. Thank you!